larray_editor/editor.py

```diff
@@ -277,6 +280,23 @@ def setup_menu_bar(self):
         help_menu.addAction(create_action(self, _('Online Objects and Functions (API) &Reference'),
                                           triggered=self.open_api_documentation))
 
+    def data_changed(self):
+        self.unsaved_modifications = True
+
+    @property
+    def unsaved_modifications(self):
+        return self._unsaved_modifications
+
+    @unsaved_modifications.setter
+    def unsaved_modifications(self, unsaved_modifications):
+        self._unsaved_modifications = unsaved_modifications
+        # add '*' at the end of the title if unsaved modifications. Remove it if not.
```

Comment:
I would rather have this setter call update_title then have update_title construct the title accordingly.

Reply:
This is what I implemented first but I thought it was an overkill. 
I guess update_title does not take so much computation time.